toml/_dotty.py

```diff
@@ -112,14 +119,23 @@ def __setitem__(self, key: str, value: Any):
             raise KeyError(f"Using '{self._BASE_DICT}' as key is not supported")
 
         keys, last = self.split(key)
+        del key
         global_key = ""
 
         item = self._data
         for k in keys:
-            global_key += f".{k}"
+            global_key += "." + str(k)
+            """
+            fstrings are cool but slow and leave garbage mem
+            For a single thing, it's best to just use +
+
+            I do not know if k is always a string.
+            if it is, remove the typecast.
+            """
             item = self._get_or_create(item, k, global_key)
 
-        item[last] = value
+        item[last] = value  # Unsure what happens here, leaving untouched.
```

Comment:
I dont know what you mean with this comment. Function iterates all parts **but last one** of the key, after splitting it at dots (ie: nesting markers).

This way, `item` is the "target" dict where we wanted to add the value, and `last` is the last part of the key, which we didnt use before

Reply:
`item` is not returned or stored. I don't know if it is safe to `del`.

Reply:
`item` should be a reference to the dict where we stored a value, thus should be safe to `del item` (the real structure still exists, but we remove this "pointer" to it)